scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -311,11 +316,33 @@ class FossId(
                     }
                 }
 
-                checkScan(scanCode)
-                val rawResults = getRawResults(scanCode)
-                val resultsSummary = createResultSummary(startTime, rawResults)
+                if (waitForResult) {
+                    checkScan(scanCode)
+                    val rawResults = getRawResults(scanCode)
+                    val resultsSummary = createResultSummary(startTime, rawResults)
 
-                results.getOrPut(pkg) { mutableListOf() } += resultsSummary
+                    results.getOrPut(pkg) { mutableListOf() } += resultsSummary
+                } else {
+                    val issue = OrtIssue(
+                        source = pkg.id.toCoordinates(),
+                        message = "This package has been scanned in asynchronous mode. Scan results are available " +
+                                "on the FossID instance.",
+                        severity = Severity.HINT
+                    )
+                    val summary = ScanSummary(
+                        startTime = startTime,
+                        endTime = Instant.now(),
+                        fileCount = 0,
+                        packageVerificationCode = "",
+                        licenseFindings = sortedSetOf(),
+                        copyrightFindings = sortedSetOf(),
+                        issues = listOf(
+                            issue
+                        )
+                    )
+
+                    results.getOrPut(pkg) { mutableListOf() } += ScanResult(UnknownProvenance, details, summary)
```

Comment:
The `UnknownProvenance` here strikes me. I know it has been like this before in `createResultSummary()`, but *why* is the provenance actually unknown currently in any case? We *do* know what's been scanned, as we told FOSSID what to scan, or?

Reply:
Yes this is right. We can replace it by `RepositoryProvenance` right ?
Does it needs to be a separate commit ?